test/platform-stats.test.js

```diff
@@ -0,0 +1,78 @@
+import assert from 'node:assert'
+import pg from 'pg'
+import { beforeEach, describe, it } from 'mocha'
+
+import { DATABASE_URL } from '../lib/config.js'
+import { migrateWithPgClient } from '../lib/migrate.js'
+import { VALID_MEASUREMENT, VALID_STATION_ID } from './helpers/test-data.js'
+import { updateDailyNodeMetrics } from '../lib/platform-stats.js'
+
+const createPgClient = async () => {
+  const pgClient = new pg.Client({ connectionString: DATABASE_URL })
+  await pgClient.connect()
+  return pgClient
+}
+
+describe('platform-stats', () => {
+  let pgClient
+  before(async () => {
+    pgClient = await createPgClient()
+    await migrateWithPgClient(pgClient)
+  })
+
+  let today
+  beforeEach(async () => {
+    await pgClient.query('DELETE FROM daily_node_metrics')
+
+    // Run all tests inside a transaction to ensure `now()` always returns the same value
+    // See https://dba.stackexchange.com/a/63549/125312
+    // This avoids subtle race conditions when the tests are executed around midnight.
+    await pgClient.query('BEGIN TRANSACTION')
+    today = await getCurrentDate()
+  })
+
+  afterEach(async () => {
+    await pgClient.query('END TRANSACTION')
+  })
+
+  after(async () => {
+    await pgClient.end()
+  })
+
+  describe('updateDailyNodeMetrics', () => {
+    it('updates daily node metrics for today with multiple measurements', async () => {
+      const validStationId2 = VALID_STATION_ID.slice(0, -1) + '1'
+      const honestMeasurements = [
+        { ...VALID_MEASUREMENT, stationId: VALID_STATION_ID },
+        { ...VALID_MEASUREMENT, stationId: validStationId2 }
+      ]
+
+      await updateDailyNodeMetrics(pgClient, honestMeasurements)
+
+      const { rows } = await pgClient.query('SELECT station_id, day::TEXT FROM daily_node_metrics ORDER BY station_id')
```

Comment:
This line is very long; please break it into multiple shorter ones. See how we are building multi-line SQL statements in existing code.